[AArch64] Turn MaxInterleaveFactor into a subtarget feature

davemgreen opened the pull request:
The default value for MaxInterleaveFactor is 2, but some CPUs prefer a wider factor of 4. This adds a subtarget feature so that cpus can override the default in their tuning features, keeping more of the options together in one place.

fhahn reviewed (approved):
LGTM, thanks.

This presumably also allows specifying the target feature to use max interleave factor 4 for CPUs that do not have it (or disable it via the target-feature if for CPUs that do). MIght be good to add a test that just uses the target feature?

davemgreen committed:
[AArch64] Turn MaxInterleaveFactor into a subtarget feature

The default value for MaxInterleaveFactor is 2, but some CPUs prefer a wider
factor of 4. This adds a subtarget feature so that cpus can override the
default in their tuning features, keeping more of the options together in one
place.
